modules/opensearch/opensearch.client.inc

```diff
@@ -92,6 +92,13 @@ function opensearch_get_objects(array $ids, $local = FALSE) {
     $request->setProfile($profile);
   }
 
+  if ($local) {
+    // Ensure local ids are integer values and not strings. Else we will not
+    // hit the cache, since they will be converted to integers in our caching
+    // logic in opensearch_execute(), if they are passed as strings.
+    $ids = array_map('intval', $ids);
```

Comment:
I wonder if this the right level for addressing such an issue or it should be done lower down the call stack.

Reply:
I actually found out that they need to be strings. More in comment below and in code/commit messages.